core-server/src/main/java/org/glassfish/jersey/server/internal/monitoring/core/ReservoirConstants.java

```diff
@@ -16,28 +16,55 @@
 
 package org.glassfish.jersey.server.internal.monitoring.core;
 
+import java.security.PrivilegedAction;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.glassfish.jersey.server.ServerProperties;
+
 /**
  * The constants that determine the behaviour of sliding windows and their trimmers.
  *
  * @author Stepan Vavra
  */
 public final class ReservoirConstants {
 
+    private static final int DEFAULT_COLLISION_BUFFER_POWER = 8;
+
     /**
      * Allow for 2^that many duplicate ticks before throwing away measurements.
+     * This value is by default {@link DEFAULT_COLLISION_BUFFER_POWER}, but it can be configured
+     * with {@link COLLISION_BUFFER_POWER_JVM_ARG} JVM argument
      */
-    public static final int COLLISION_BUFFER_POWER = 8;
+    public static final int COLLISION_BUFFER_POWER;
 
     /**
      * The size of the collision buffer derived from the collision buffer power.
      */
-    public static final int COLLISION_BUFFER = 1 << COLLISION_BUFFER_POWER; // 256
+    public static final int COLLISION_BUFFER;
 
     /**
      * Only trim on updating once every N.
      */
     public static final int TRIM_THRESHOLD = 256;
 
+    static {
+        PrivilegedAction<Integer> action = new PrivilegedAction<Integer>() {
+            @Override
+            public Integer run() {
+                return Integer.getInteger(ServerProperties.COLLISION_BUFFER_POWER_JVM_ARG, DEFAULT_COLLISION_BUFFER_POWER);
+            }
+        };
+        int collisionBuffePower = DEFAULT_COLLISION_BUFFER_POWER;
+        try {
+            collisionBuffePower = action.run();
```

Comment:
actually I'm not sure this will work in favor of ``AccessController.doPrivileged(action);`` because purpose of the AccessController is to provide to privileged action those privileges which previously that action did not have. So, AccessControler shall supply those 
```
 // Needed by TimeWindowStatisticsImplTest
  permission java.util.PropertyPermission "jersey.config.server.monitoring.collision.buffer.power", "read,write";
  ```
  you are providing for test purposes to the privileged action of getting property. While securityManager is on for sure. 
  And running action just by ``run()`` method will just invoke it as usual method without any additional privileges. 
  And SecurityException might not be caught when using  AccessController because in that case user can supply proper privileges to be provided to action and thus SecurityException won't be thrown. On the other hand when you just run the action and exception occurs there is no way for privileges to be supplied to it so it will always fail and application will always use default collision buffer.